bench/bm_case1.cpp

```diff
@@ -26,8 +26,8 @@ class math_op : public fg::node<math_op<T, op>, fg::IN<T, 0, N_MAX, "in">, fg::O
     explicit math_op(T factor, std::string name = fair::graph::this_source_location()) : _factor(factor) { this->set_name(name); }
 
     template<fair::meta::t_or_simd<T> V>
-    [[nodiscard]] constexpr auto
-    process_one(V a) const noexcept {
+    [[nodiscard, gnu::always_inline]] constexpr auto
```

Comment:
Is the `gnu::always_inline` strictly required to enable SIMD? From a user-API perspective, this is a bit odd and dangerous because it suggests that w/o forced inline the SIMD and process_one-unrolling wouldn't work. I'd prefer to keep the forced inlining reserved for library internal things and only where we know that it's necessary. Sometimes forced inlining can have a also have the opposite effect that the code size becomes too large w.r.t. fitting into the L1/L2 cache...

Reply:
It's not required to enable SIMD. It simply helps the compiler to *not* make stupid inliner decisions which can hurt performance. I'm fine with removing them and blaming an N-times slowdown on the compilers bad quality-of-implementation.